x/evm/vm/geth/geth.go

```diff
@@ -55,20 +69,24 @@ func (e EVM) Config() vm.Config {
 // and the current chain configuration. If the contract cannot be found it returns
 // nil.
 func (e EVM) Precompile(addr common.Address) (p vm.PrecompiledContract, found bool) {
-	precompiles := GetPrecompiles(e.ChainConfig(), e.EVM.Context.BlockNumber)
-	p, found = precompiles[addr]
+	p, found = e.precompiles[addr]
 	return p, found
 }
 
 // ActivePrecompiles returns a list of all the active precompiled contract addresses
 // for the current chain configuration.
-func (EVM) ActivePrecompiles(rules params.Rules) []common.Address {
-	return vm.ActivePrecompiles(rules)
+func (e EVM) ActivePrecompiles(rules params.Rules) []common.Address {
+	// TODO e.precompiles
```

Comment:
Yes, this `ActivePrecompiles` is not actually used now. I changed it when I was testing and forgot to revert the changes. For clarity, I will remove them.

Reply:
This requires a geth fork right?

Reply:
Ethermint uses its own `ActivePrecompiles` function in `state_transition.go`. The only other places I've seen `vm.ActivePrecompiles` used in go-ethereum is tracer-related.